services/wireguard/service/service_windows.go

```diff
@@ -131,6 +143,8 @@ func (manager *Manager) Serve(providerID identity.Identity) error {
 func (manager *Manager) Stop() error {
 	manager.wg.Done()
 
+	manager.cleanup()
```

Comment:
this introduces a race condition. The cleanup might still be nil if I call serve stop in rapid succession. Need to put this under a lock with a proper check.

Reply:
Changed it to be deferred right after creation.